global/scripts/fsl_version.shlib

```diff
@@ -10,20 +16,162 @@
+    local fsl_version
+    local __functionResultVar=${1}
+
+    fsl_version_file="${FSLDIR}/etc/fslversion"
+
+    if [ -f ${fsl_version_file} ]
+    then
+        fsl_version=`cat ${fsl_version_file}`
+        log_Msg "INFO: Determined that the FSL version in use is ${fsl_version}"
+    else
+        log_Msg "ERROR: Cannot tell which version of FSL you are using."
+        exit 1
+    fi
+
+    eval $__functionResultVar="'${fsl_version}'"
+}
+
+#
+# Function Description
+#  Standardize the FSL-version number (in argument 1) for easy
+#  version check and comparison
+#  
+#  This function take a version number (e.g. 6.0.4)
+#  and format it as Num.Num.Num.Num (e.g. 6.0.4.0). 
+#  This make the version comparison/check easier 
+#  (e.g. when comparing 6.0.4.0 with 6.0.4.1). 
+#  Here we assume that FSL-version is not using more than
+#  4 digits and 3 dots (M.X.Y.Z), otherwise increase max_ndots. 
+#
+#  Some formatting examples:
+#  6 => 6.0.0.0 
+#  6.0 => 6.0.0.0
+#  6.0.4 => 6.0.4.0
+#  6.0.4.12 => 6.0.4.12 
+#
+# Usage Example:
+#  fsl_ver2=$(fsl_version_format "6.0.4")
+#
+fsl_version_format(){
+    local fv="${1}"
+    local ndots
+    # maximum number of dots in FSL-version string
+    local max_ndots=$MAX_NDOTS_IN_FSLVERSION
+
+    ndots=$(awk -F"." '{print NF-1}' <<< "$fv")
+
+    while [ "$ndots" -lt "$max_ndots" ]; do
+        fv="$fv".0
+        ndots=$(awk -F"." '{print NF-1}' <<< "$fv")
+    done
+    
+    echo "$fv"
 }
+
+# Function Description
+#  Check if the current FSL version (the FSL version in use) 
+#  is at least equal to the version given in argument 1 ($1). 
+#
+#  This function echoes string "true" (if FSL-version >= argument1) 
+#  or "false" (if FSL-version < argument1).
+#  It is up to the user to decide what to do with this information. 
+# 
+# Usage Example:
+#  ret_val=$(fsl_minimum_required_version "6.0.7.2")
+#  if [ "$ret_val" == "false" ]; then
+#      log_Msg "ERROR: FSL minimun required version is 6.0.7.2. 
+#      exit 1
+#  else 
+#      log_Msg "INFO: FSL version > 6.0.7.2"
+#      echo "do something..."
+#  fi 
+# 
+fsl_minimum_required_version(){
+    local min_ver="${1}" 
+    local cur_ver 
+    local min_ver_array
+    local cur_ver_array
+    # maximum number of dots in FSL-version string
+    local max_ndots=$MAX_NDOTS_IN_FSLVERSION
+
+    # get current FSL version, format it in X.Y.Z.W, format min version too
+    fsl_version_get cur_ver >> /dev/null
+    cur_ver=$(fsl_version_format "$cur_ver")
+    min_ver=$(fsl_version_format "$min_ver")
+
+    # if the 2 version strings are identical, no need to test further
+    if [ "$cur_ver" == "$min_ver" ]; 
+    then
+        echo "true"
+        return 0
+    fi
+
+    # split version by dots - assuming the format X.Y.Z.W
+    for k in $(seq 0 $max_ndots); 
+    do 
+        cur_ver_array[$k]=$(awk -F"." -v i=$((k+1)) '{print $i}' <<< "$cur_ver")
+        min_ver_array[$k]=$(awk -F"." -v i=$((k+1)) '{print $i}' <<< "$min_ver")
+    done
```

Comment:
```suggestion
    IFS=. read -a cur_ver_array <<< "$cur_ver"
    IFS=. read -a min_ver_array <<< "$min_ver"
```

You could consider using the array lengths in the test loop to handle the versions as-is, rather than preformatting.

Reply:
In this case, we also need to consider arbitrary length and find the maximum number of dots from `$cur_ver`and `$min_ver` for the subsequent comparison (otherwise it won't work with more that 3 dots). This will make things much more generic (and the comparison via fsl_minimum_required_version function will work any FSL-version string length).  
Great suggestions. 
I did that and the test results are: 
```
==> Test variable and unsual FSL string length
==> Test fsl_version_format on unsual FSL string length
Test fsl_version_format 6 return 6.0.0.0
Test fsl_version_format 7.1.2.0.1 return 7.1.2.0.1
Test fsl_version_format 6.0.7.6.0 return 6.0.7.6.0
Test fsl_version_format 6.0.7.6.1 return 6.0.7.6.1
Test fsl_version_format 6.0.7.6.0.0 return 6.0.7.6.0.0
Test fsl_version_format 6.0.7.6.0.1 return 6.0.7.6.0.1
====
current FSL-Version = 6.0.7.6
==> Test function fsl_minimum_required_version on unsual FSL string length: 
Test: minimum-ver <= current-ver is ? 
Test:               6 <= 6.0.7.6 is true
Test:       7.1.2.0.1 <= 6.0.7.6 is false
Test:       6.0.7.6.0 <= 6.0.7.6 is true
Test:       6.0.7.6.1 <= 6.0.7.6 is false
Test:     6.0.7.6.0.0 <= 6.0.7.6 is true
Test:     6.0.7.6.0.1 <= 6.0.7.6 is false 
```
I upload the modify version of fsl_version.shlib to resolve the issue.

Reply:
I Uploaded the modification for arbitrary length of FSL-version string. 
The consequence, is the `fsl_version_format()` and the constant `MAX_NDOTS_IN_FSLVERSION=3` are just for cosmetic. I left these functions.
Do you want to remove them ?